src/components/views/auth/InteractiveAuthEntryComponents.tsx

```diff
@@ -320,8 +320,20 @@ export class TermsAuthEntry extends React.Component<ITermsAuthEntryProps, ITerms
 
     componentDidMount() {
```

Comment:
It should be possible to keep the listener in a single location in the Registration component, and then add and remove it on the fly as registration moves through the different stages using `componentDidUpdate`

Reply:
I am not sure i really understand this clearly, as requested by @t3chguy the warning should only be available at a later stage of the registration which is fine but according to your suggestion now, is it possible to actually detect if it has moved from one stage to another since all the stages are just having one route `/register`?.

Reply:
Yes, the Registration component holds state that determines which stage of the registration process the user is on, so we should be able to detect in that component when the user has moved past the first stage

Reply:
I need a bit of your help here, currently the state variable i think keeps track of the stages is `flow` which is an array of object containing a property name `stages` which is also an array,the issue here is that i tried using componentDidUpdate and log to the console to check  if there is a state change btwn previous and current so i can leverage on that but unfortunately there isn't, also immediately i hit the register button, it stops logging the current and prev values and rather logs the following instead, which i couldn't find anywhere in the project

![console](https://user-images.githubusercontent.com/38439166/164701525-4eb10b87-4adc-4d11-aee2-fc419b4a501b.PNG)



This is when it still logs the curr and prev: 

![console2](https://user-images.githubusercontent.com/38439166/164701806-5eedb022-b26f-47c9-9f38-f1442f513054.PNG)

Reply:
Also lastly, in my opinion including the listeners in both components as i have done previously doesn't hurt atall because throughout the entire application it's only the Auth section that makes use of this components, even if it were to be used by another component, it still serves it's purpose of not allowing user refresh without warning them of a data loss.

Reply:
Nonetheless, i'll appreciate if you guys can point me in the right direction.

Reply:
I think just conditioning on `this.state.doingUIAuth` should be fine

> it's only the Auth section that makes use of this components

Auth yes, but not Registration. The issue was about Registration, not changing password/deleting devices/deactivating account which can all also run the UIA code

Reply:
Ok boss, thanks... Will try to work things out as you have suggested.